public/resources/js/form.js

```diff
@@ -1,76 +1,89 @@
 
-    pass.addEventListener('input',() => {
-        if(pass.value.length > 0){
-            msg.style.display = "block";
-        }else{
-            msg.style.display = "none";
+pass.addEventListener('input',() => {
+    if(pass.value.length > 0){
+        msg.style.display = "block";
+        for (let i = 0; i < icon.length; i++) {
+            icon[i].style.top = "47px";
         }
-        if(pass.value.length < 4){
-            str.innerHTML = "weak";
-            pass.style.borderColor = "#b22323";
-            msg.style.color = "#b22323";
-        } else if(pass.value.length >= 4 && pass.value.length < 8){
-            str.innerHTML = "medium";
-            pass.style.borderColor = "#bc5806";
-            msg.style.color = "#bc5806";
-        } else if(pass.value.length >= 8){
-            str.innerHTML = "strong";
-            pass.style.borderColor = "#356212";
-            msg.style.color = "#356212";
-        }
-        checkPasswordConfirmation();
-    });
+        
+    }else{
+        msg.style.display = "none";
+        pass.style.backgroundColor = "rgba(255,255,255,0.2)";
+    }
+    if(pass.value.length > 0 && pass.value.length < 4){
+        str.innerHTML = "weak";
+        pass.style.backgroundColor = "rgba(211, 12, 12, 0.486)";
+        msg.style.color = "#b11010";
+        msg.style.marginBottom = "10px";
+        msg.style.top = "-8px";
+        msg.style.position = "relative";
+    } else if(pass.value.length >= 4 && pass.value.length < 8){
+        str.innerHTML = "medium";
+        pass.style.backgroundColor = "rgba(236, 240, 22, 0.486)";
+        msg.style.color = "#f3eb51";
+        msg.style.marginBottom = "10px";
+        msg.style.top = "-8px";
+        msg.style.position = "relative";
+    } else if(pass.value.length >= 8){
+        str.innerHTML = "strong";
+        pass.style.backgroundColor = "rgba(81, 182, 14, 0.651)";
+        msg.style.color = "#5bec64";
+        msg.style.marginBottom = "10px";
+        msg.style.top = "-8px";
+        msg.style.position = "relative";
+    }
+    checkPasswordConfirmation();
+});
 
-    copass.addEventListener('input',() =>{
-        checkPasswordConfirmation();
-    });
+copass.addEventListener('input',() =>{
+    checkPasswordConfirmation();
+});
 
-    email.addEventListener('input',() =>{
-        validateEmail();
-    });
-    
-    function checkPasswordConfirmation() {
-        if(copass.value.length > 0){
-            nmsg.style.display = "block";
-        }else{
-            nmsg.style.display = "none";
-            return;
-        }
-        if(copass.value === pass.value ){
-            nostr.innerHTML = "matches";
-            copass.style.borderColor = "#356212";
-            nmsg.style.color = "#356212";
-        }else{
-            nostr.innerHTML = "does not match";
-            copass.style.borderColor = "#b22323";
-            nmsg.style.color = "#b22323";
-        }
-    };
+email.addEventListener('input',() =>{
+    validateEmail();
+});

```

Comment:
if you are giving up in generalizing, to make new code each time a new feature is assigned to you then it is okay because this will work with just one view. This is because you are attaching a anonym function to an addeventlister to a specific element, if you did the function instead of attaching to events and call it from the html as oninput attribute will result in more opportunities to resolve future problems, not naming functions that could be used in more context saving time to the future you could make you regret in some point.